input-profile.ts

```diff
@@ -0,0 +1,277 @@
+import {Component, Object3D, Emitter} from '@wonderlandengine/api';
+import {HandTracking} from './hand-tracking.js';
+import {vec3, quat} from 'gl-matrix';
+import {property} from '@wonderlandengine/api/decorators.js';
+import {VrModeActiveSwitch} from './vr-mode-active-switch.js';
+
+const _TempVec = vec3.create();
+const _TempQuat = quat.create();
+const _TempRotation1 = new Float32Array(4);
+const _TempRotation2 = new Float32Array(4);
+const minTemp = new Float32Array(3);
+const maxTemp = new Float32Array(3);
+const hands = ['left', 'right'];
+
+interface VisualResponse {
+    target: Object3D;
+    min: Object3D;
+    max: Object3D;
+    id: number;
+}
+
+export class InputProfile extends Component {
+    static TypeName = 'input-profile';
+    static Cache: Map<string, any> = new Map();
+    private _gamepadObjects: Record<string, Object3D> = {};
+    private _controllerModel: Object3D | null = null;
+    private _defaultControllerComponents: Component[] | undefined;
+    private _handedness!: string;
+    private _profileJSON: any = null;
+    private _gamepad: Gamepad | undefined;
+    private _buttons: VisualResponse[] = [];
+    private _axes: VisualResponse[] = [];
+
+    urlEmitter: Emitter = new Emitter();
```

Comment:
Rename that `onModelLoaded` since it's basically a generic emitter stating that things are ready